### PR TITLE
add contextual search for selected item

### PR DESCRIPTION
**Pull request by peterp**
for issue #407, still trying to figure out how to test this, any tips?

I originally thought I could target a detailed selector with a keymap: `.tree-view li.directory,` but I couldn't get that to work. I need to dive a bit deeper into `atom-keymap` to figure out why.

### CODE REVIEW

**Comment by benogle** [comment]
Can you write specs for this PR?

**Comment by peterp** [comment]
@benogle I can, I just need some direction. My initial was to test for the visibility of a ".search/ .search-project" class, but further inspection into the specs made me realise that this isn't loaded into tree-view's specs. (And maybe shouldn't be?)

Is there a precedent for writing specs that dispatch atom commands?

**Comment by lee-dohm** [comment]
I would start with checking for the visibility like you were first planning to do and see if you can make that robust. If that doesn't work or doesn't work reliably, let us know!

**Comment by peterp** [comment]
@lee-dohm I've added the tests that check for visibility, but the .find-and-replace and .find-project elements are null, do I have to activate the package?

**Comment by lee-dohm** [comment]
Yes, you have to manually activate any packages you want to work in specs.

**Comment by peterp** [comment]
@lee-dohm I've now added the `find-and-replace` package, but the tests still don't find the class names.

File: spec/tree-view-spec.coffee [hunk 1]
```diff
@@ -2530,6 +2534,43 @@ describe "TreeView", ->
               expect(treeView.list).toHaveClass('full-menu')
               expect(treeView.list).not.toHaveClass('multi-select')
 
+  fdescribe "search based on selected items", ->
```
Comment (liuderchi) [inline spec/tree-view-spec.coffee]: Typo: `fdescribe`

**Comment by tenorok** [comment]
Hello, is there any chance for merging?

**Comment by peterp** [comment]
@lee-dohm this is ready for review again, an unrelated test is failing, but passing locally.

**Comment by lee-dohm** [comment]
Thanks @peterp! I'm swamped today and tomorrow, but I'll take a look Wednesday :+1:

**Comment by lee-dohm** [comment]
@peterp Sorry I haven't gotten back to you on this yet. It is still on my radar. I'm working on an improvement for it as well. I'll touch base on it early next week.

**Comment by lee-dohm** [comment]
@peterp In digging deeper on this, I found that the [find-and-replace package has context menu items that it adds to the tree-view](https://github.com/atom/find-and-replace/blob/master/menus/find-and-replace.cson#L24-L27). A similar approach could be used to add the keybindings you're wanting here and I bet the change would be much simpler.

What do you think?

**Comment by peterp** [comment]
@lee-dohm Sorry, I've had a bit of a vacation, that sounds like a much better approach, and we can keep the functionality where it belongs. I'll close this off and get back to you.

**Comment by lee-dohm** [comment]
Sounds good. Let me know if you need any help with it.

**Pull request closed without merging**